src/cdrizzlebox.c

```diff
@@ -26,18 +26,21 @@ inline_macro static int
 update_data(struct driz_param_t* p, const integer_t ii, const integer_t jj,
             const float d, const float vc, const float dow) {
 
-  const double vc_plus_dow = vc + dow;
+  double vc_plus_dow;
   
-  /* Just a simple calculation without logical tests */
-  if (vc == 0.0) {
+  if (dow == 0.0f) return 0;
+  
+  vc_plus_dow = vc + dow;
+  
+  if (vc == 0.0f) {
```

Comment:
Being C, wouldn't this potentially get tripped up by floating point error?

Reply:
No (not for our our purpose). 0.0 is 0.0 exactly. So if some pixel in the output weight array (that is what `vc` really is) has value exactly 0.0 - then all it means that we can skip the fancy (complete) computation done in the `else` block below and just assign `dow` (input pixel weight) to `vc` (output pixel weight). If "somehow" (due to floating errors or "for real")`vc !=0` then the code will proceed with full computations in the `else` block. Nothing dangerous here.

Reply:
No (not for our purpose). 0.0 is 0.0 exactly. So if some pixel in the output weight array (that is what `vc` really is) has value exactly 0.0 - then all it means that we can skip the fancy (complete) computation done in the `else` block below and just assign `d` (input pixel) to the output array (`output_data`) without doing weighted sum and assign `vc` (output pixel weight) to the output weight array (`output_counts`) at the end of the code. If "somehow" (due to floating errors or "for real")`vc !=0` then the code will proceed with full computations in the `else` block. Nothing dangerous here.